core/src/main/java/com/scalar/db/exception/transaction/TransactionException.java

```diff
@@ -1,13 +1,44 @@
 package com.scalar.db.exception.transaction;
 
+import java.util.Optional;
+import javax.annotation.Nullable;
+
 /** Base class for all exceptions thrown by the Transaction API. */
 public class TransactionException extends Exception {
 
-  public TransactionException(String message) {
-    super(message);
+  @Nullable private final String transactionId;
+
+  public TransactionException(String message, @Nullable String transactionId) {
+    super(addTransactionIdToMessage(message, transactionId));
+    this.transactionId = transactionId;
+  }
+
+  public TransactionException(String message, Throwable cause, @Nullable String transactionId) {
+    super(addTransactionIdToMessage(message, transactionId), cause);
+    this.transactionId = transactionId;
   }
 
-  public TransactionException(String message, Throwable cause) {
-    super(message, cause);
+  /**
+   * Returns a transaction ID that associated with the transaction whose status is unknown.
+   *
+   * @return a transaction ID that associated with the transaction whose status is unknown
+   */
+  public Optional<String> getTransactionId() {
+    return Optional.ofNullable(transactionId);
+  }
+
+  private static String addTransactionIdToMessage(String message, @Nullable String transactionId) {
```

Comment:
I added this method to add a transaction ID to the message based on Vincent's idea. The constructors convert the message with the method and pass it to the constructors of the superclass.